### PR TITLE
chore: implemented dynamic loading of translations

### PR DESCRIPTION
**Pull request by Abhikumar98**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,5 +33,9 @@
   "workspaces": [
     "packages/*"
   ],
-  "packageManager": "yarn@3.6.1"
+  "packageManager": "yarn@3.6.1",
+  "dependencies": {
```
Comment (Abhikumar98) [inline package.json:37]: this will be moved to respective package

**Comment by Abhikumar98** [comment]
> 

Sure, adding it

**Comment by Abhikumar98** [comment]
> Please check this out. When we load the page there is no fallback and we see plain keys. Can we make english as fallback language that is always there?
> 
>  Screen.Recording.2023-08-14.at.11.43.37.AM.mov

Hey @chybisov,
`en` is already set as the fallback language here, but since we are importing all the languages dynamically it takes sometime to load the fallback as well, here the fallback is working as when there is no language selected.

Although in our case we want to have a "pre-loaded" language, which will be already imported and loaded prior to lazy loading of languages and this isn't working as expected:

I've tried:
1. Importing the en.json file but to no avail, as it still picks the file loaded from the backend resource
2. Supplying just the fallback language to resource, expecting it to later "lazily" add more languages to resources but this isn't working as expected as well.

But this latency in languages being loaded is expected imo, but lemme know if there's a possible solution that I've missed out.

Thanks !

**Comment by Abhikumar98** [comment]
> > Please check this out. When we load the page there is no fallback and we see plain keys. Can we make english as fallback language that is always there?
> > Screen.Recording.2023-08-14.at.11.43.37.AM.mov
> 
> Hey @chybisov, `en` is already set as the fallback language here, but since we are importing all the languages dynamically it takes sometime to load the fallback as well, here the fallback is working as when there is no language selected.
> 
> Although in our case we want to have a "pre-loaded" language, which will be already imported and loaded prior to lazy loading of languages and this isn't working as expected:
> 
> I've tried:
> 
> 1. Importing the en.json file but to no avail, as it still picks the file loaded from the backend resource
> 2. Supplying just the fallback language to resource, expecting it to later "lazily" add more languages to resources but this isn't working as expected as well.
> 
> But this latency in languages being loaded is expected imo, but lemme know if there's a possible solution that I've missed out.
> 
> Thanks !

Update:

Tried a config of partially loading the resources from backend and using static ones, seems to be working for now.

File: packages/dapp/src/providers/I18nProvider.tsx [hunk 2]
```diff
@@ -5,10 +5,19 @@ import React, { PropsWithChildren, useMemo } from 'react';
...
 export const I18NProvider: React.FC<PropsWithChildren<{}>> = ({ children }) => {
   const languageMode = useSettingsStore((state) => state.languageMode);
 
+  const resources = {
+    en: {
+      translation: {
+        ...en,
+      },
+    },
+  };
```
Comment (chybisov) [inline packages/dapp/src/providers/I18nProvider.tsx]: I think this works fine as well

```ts
const resources = {
    en: {
      translation: en,
    },
  };
```

Or we can even make it inline in the init function below

File: packages/dapp/src/const/MenuConfig/useLanguagesContent.tsx [hunk 3]
```diff
@@ -23,12 +24,14 @@ export const useLanguagesContent = () => {
     });
   };
 
-  const languages = Object.keys(i18n.store.data)
+  console.log({ data: i18n.store.data });
```
Comment (chybisov) [inline packages/dapp/src/const/MenuConfig/useLanguagesContent.tsx:27]: 🙂

**Review by chybisov — CHANGES_REQUESTED** [review]

**Comment by chybisov** [comment]
@Abhikumar98, the last update is really nice and works well. Please see a couple of comments above 🙂

After we merge this I'll update crowdin configuration to a new file structure and we can proceed with updating widget with the same config 🙂

**Comment by Abhikumar98** [comment]
hey @chybisov,
All the changes mentioned above are done, let's have a last review and merge if it's good to go :)

**Review by chybisov — APPROVED** [review]

File: packages/dapp/package.json [hunk 4]
```diff
@@ -20,7 +20,9 @@
     "@widgetbot/react-embed": "^1.8.0",
     "ethers": "^5.7.2",
     "i18next": "^23.4.4",
-    "i18next-browser-languagedetector": "^7.1.0",
+    "i18next-browser-languagedetector": "^7.0.2",
```
Comment (dennyscode) [inline packages/dapp/package.json:23]: Was it necessary to downgrade this package? So, should we keep this in mind when bumping our packages in order to not break anything?

**Review by dennyscode — APPROVED** [review]

File: packages/dapp/package.json (hunk 4 above)
Comment (chybisov) [reply packages/dapp/package.json:23]: @dennyscode, nice catch. No, this shouldn't be downgraded 🙂

**Review by chybisov — CHANGES_REQUESTED** [review]
Please upgrade i18next packages.

**Review by chybisov — APPROVED** [review]

**Pull request merged**